### PR TITLE
PROGRAMMES-7085 - update Symfony to 4.3

### PR DESCRIPTION
**Pull request by Bobkidbob**
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-7085

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: tests/Controller/Schedules/VanityRedirectControllerTest.php [hunk 1]
```diff
@@ -32,7 +33,7 @@ public function testUserIsRedirected(string $vanity, string $urlSuffix)
 
         $client->request('GET', $url . '?utcoffset=%2B01%3A00');
 
-        $this->assertRedirectTo($client, 302, '/schedules/p00rfdrb/' . $urlSuffix . '?utcoffset=%2B01%3A00');
+        $this->assertRedirectTo($client, 302, '/schedules/p00rfdrb/' . $urlSuffix . '?utcoffset=%2B01:00');
```
Comment (gmarinov) [inline tests/Controller/Schedules/VanityRedirectControllerTest.php]: why?

File: src/Translate/PoFileLoaderExceptItActuallyWorks.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Translate;
+
+use Symfony\Component\Translation\Loader\PoFileLoader;
+
+class PoFileLoaderExceptItActuallyWorks extends PoFileLoader
```
Comment (gmarinov) [inline src/Translate/PoFileLoaderExceptItActuallyWorks.php]: would love a comment why this works this way. bonus if ticket number or GitHub link.

Comment (gmarinov) [reply src/Translate/PoFileLoaderExceptItActuallyWorks.php]: p.s. what's a better name for this?

File: tests/Controller/Schedules/VanityRedirectControllerTest.php (hunk 1 above)
Comment (gmarinov) [inline tests/Controller/Schedules/VanityRedirectControllerTest.php]: why?

File: src/Translate/PoFileLoaderExceptItActuallyWorks.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+{
+    protected function loadResource($resource)
```
Comment (BjornTwachtmann) [inline src/Translate/PoFileLoaderExceptItActuallyWorks.php]: Add a comment here saying why this insane thing exists please :)

File: src/Translate/Translator.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+namespace App\Translate;
+
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Symfony\Contracts\Translation\TranslatorTrait;
+
+class Translator implements TranslatorInterface
+{
+    use TranslatorTrait;
+
+    private function getPluralizationRule(int $number, string $locale): int
```
Comment (BjornTwachtmann) [inline src/Translate/Translator.php]: Bit of a comment on what this does would be useful too

File: tests/DsAmen/Presenters/Domain/CoreEntity/CollapsedBroadcast/CollapsedBroadcastPresenterTest.php [hunk 5]
```diff
@@ -45,7 +45,7 @@ class CollapsedBroadcastPresenterTest extends TestCase
     public function setUp()
     {
         $this->mockRouter = $this->createMock(UrlGenerator::class);
-        $this->translate = new TranslateProvider(new TranslateFactory());
+        $this->translator = Translator::new();
```
Comment (gmarinov) [inline tests/DsAmen/Presenters/Domain/CoreEntity/CollapsedBroadcast/CollapsedBroadcastPresenterTest.php]: not e.g. `new Translator()` ?

**Pull request merged**